### ISSUE TITLE
[BUG] Fresh install fails on nightly on MacOS

### ISSUE THREAD

**Issue by kraemahz** [amethyst/amethyst#2053]
## Description
From the latest nightly
```
rustc --version
rustc 1.41.0-nightly (412f43ac5 2019-11-24)
```

using Cargo.toml with
```
[dependencies.amethyst]
version = "*" 
git = "https://github.com/amethyst/amethyst.git"
features = ["nightly"]
```
Fails like this on MacOS:
```
cargo +nightly run --release --features metal

error[E0282]: type annotations needed
   --> /Users/kraemahz/.cargo/registry/src/github.com-1ecc6299db9ec823/gfx-backend-vulkan-0.3.3/src/window.rs:213:17
    |
213 |                 msg_send![view, setLayer: layer];
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ consider giving `result` a type
    |
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0282`.
error: could not compile `gfx-backend-vulkan`.
```

**Comment by kraemahz** [comment]
This was failing on stable too, I couldn't get anything in 0.13 to work for me with this `metal` failure. I've mostly worked around it by using @valkum's work on rendy integration in his `winit_private` branch. The `metal` feature is still picking up Vulkan dependencies through a macro though so I had to comment that out entirely (which is obviously not the fix):

```diff
diff --git a/amethyst_rendy/src/types.rs b/amethyst_rendy/src/types.rs
index 939679a3..d07f4504 100644
--- a/amethyst_rendy/src/types.rs
+++ b/amethyst_rendy/src/types.rs
@@ -114,7 +114,7 @@ impl_backends!(
     // It will be re-enabled when it actually works.
     // Dx12, "dx12", rendy::dx12::Backend; 
     Metal, "metal", rendy::metal::Backend;
-    Vulkan, "vulkan", rendy::vulkan::Backend;
+    //Vulkan, "vulkan", rendy::vulkan::Backend;
     Empty, "empty", rendy::empty::Backend;
 );
```

Now all the issues are integration failures between my code and master..

**Comment by valkum** [comment]
Yeah the msg_send issue is fixed in rendy v0.5.x I think.
But I am researching some bugs that I found when migrating to winit v0.20 so rendy v0.5 and winit v0.20 need some more time for amethyst.
The problem with metal might be fixed in the newest rendy version, but I don't encounter any crashes using my branch on macos catalina

**Comment by kraemahz** [comment]
It looks like my problems are covered by #2046 so I'm going to close this as a duplicate of that.
